src/sdk/main/src/TokenRelationship.cc

```diff
@@ -76,22 +108,44 @@ void TokenRelationship::setKycStatus(std::string_view kycStatusString)
 }
 
 //-----
-void TokenRelationship::setFreezeStatus(std::string_view freezeStatus)
+void TokenRelationship::setFreezeStatus(int freezeStatus)
 {
-  if (freezeStatus == "FROZEN")
+  if (freezeStatus == 1)
   {
     mFreezeStatus = true;
   }
-  else if (freezeStatus == "UNFROZEN")
+  else if (freezeStatus == 2)
   {
     mFreezeStatus = false;
   }
-  else if (freezeStatus == "NOT_APPLICABLE")
+  else if (freezeStatus == 0)
   {
     mFreezeStatus = std::nullopt;
   }
   else
   {
     throw std::invalid_argument("Invalid freeze status string");
   }
+}
+
+//-----
+proto::TokenKycStatus TokenRelationship::getKycStatus() const
+{
+  if (!mKycStatus.has_value())
+    return proto::TokenKycStatus::KycNotApplicable;
+  if (mKycStatus.value() == true)
+    return proto::TokenKycStatus::Granted;
+  else
+    return proto::TokenKycStatus::Revoked;
+}
```

Comment:
nit: add braces to these if statements

Reply:
Btw pushed the changes but this doesn't appear as outdated idk why. So you don't wonder why I resolved it and some of the others.